### PR TITLE
Answer Ruby Questions

### PR DESCRIPTION
**Pull request by susandaiute**

### CODE REVIEW

File: lib/diagnostic.rb [hunk 1]
```diff
@@ -5,44 +5,46 @@ class Diagnostic
   # Assume each entry in the list implements a length method that returns
   # a Fixnum.
   def longest(list)
-    # Your code goes here
+    @long = list.inject { |a, e| a.length > e.length ? a : e }
+    puts "#{@long}"
```
Comment (RDegnen) [inline lib/diagnostic.rb:9]:
``` rb
def longest(enum)
    enum.max { |a, b| a.length <=> b.length }
  end
```

File: lib/diagnostic.rb [hunk 2]
```diff
@@ -5,44 +5,46 @@ class Diagnostic
...
   end
 
   # Question 2
   # Write ruby code to return the sum of entries in the
   # list supplied (i.e. apply the `+` operator).
   # Assume each entry implements a `+` method.
   def sum(list)
-    # Your code goes here
+    @sum = list.reduce(0) { |a, e| a + e }
+    puts "#{@sum}"
```
Comment (RDegnen) [inline lib/diagnostic.rb:18]:
``` rb
def sum(enum)
    enum.reduce(&:+)
  end
```

File: lib/diagnostic.rb [hunk 3]
```diff
@@ -5,44 +5,46 @@ class Diagnostic
...
   end
 
   # Question 3
   # Write ruby code that returns true if some of the entries
   # in the list of numbers are odd
   def some_odd(numbers)
-    # Your code goes here
+    @is_odd = numbers.any? { |i| i.odd? ? true : false }
+    puts "#{@is_odd}"
```
Comment (RDegnen) [inline lib/diagnostic.rb:26]: `numbers.any?(&:odd?)`

File: lib/diagnostic.rb [hunk 4]
```diff
@@ -5,44 +5,46 @@ class Diagnostic
...
   end
 
   # Question 4
   # Write ruby code that returns true if every entry in the
   # list of numbers are even
   def every_even(numbers)
-    # Your code goes here
+    @is_even = numbers.all? { |i| i.even? ? true : false }
+    puts " #{@is_even}"
```
Comment (RDegnen) [inline lib/diagnostic.rb:34]: `numbers.all?(&:even?)`

File: lib/diagnostic.rb [hunk 5]
```diff
@@ -5,44 +5,46 @@ class Diagnostic
...
   end
 
   # Question 5
   # Write ruby code that returns an array with all of the
   # entries from the list of words capitalized (i.e. 'foo' becomes 'Foo').
   def transform(words)
-    # Your code goes here
+    @up_case = words.split(' ').map(&:capitalize).join(' ')
+    puts "#{@up_case}"
```
Comment (RDegnen) [inline lib/diagnostic.rb:42]: `words.map(&:capitalize)`

**Comment by RDegnen** [comment]
0 of 6. Do you want to go over this? You didn't have to assign things to instance variables

**Comment by susandaiute** [comment]
Yes please :(

**Pull request closed without merging**